### DISCUSSION TITLE
Significant changes on rear and front irradiance estimations with v0.10.0

### DISCUSSION THREAD

**Discussion by pasquierjb** [pvlib/pvlib-python#1823] [General]
The release note of pvlib 0.10.0 states that it: 
> Corrects an error in view factor calculations which are part of [pvlib.bifacial.infinite_sheds.get_irradiance()](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.bifacial.infinite_sheds.get_irradiance.html#pvlib.bifacial.infinite_sheds.get_irradiance). The error affects rear surface irradiance by a few W/m2.

However, when using the bifacial.infinite_sheds functions in my use cases, I get significant changes in POA irradiance on both front and rear surfaces. This is probably due to corrections on the poa_ground_diffuse but also on the poa_sky_diffuse. 

Here is an exemple:

```python
from pvlib.solarposition import get_solarposition
from pvlib.iotools.pvgis import get_pvgis_tmy
from pvlib.irradiance import get_extra_radiation
from pvlib import tracking
from pvlib.bifacial import infinite_sheds


lat, lon =  45.52, -122.67 # Portland
tmy = get_pvgis_tmy(lat, lon, map_variables=True)[0]
tmy.index = tmy.index.map(lambda dt: dt.replace(year=1990))
solpos = get_solarposition(tmy.index, lat, lon)
orientation = tracking.singleaxis(apparent_zenith = solpos['apparent_zenith'],
                                  apparent_azimuth = solpos['azimuth'],
                                  gcr=0.5
                                  )

dni_extra = get_extra_radiation(tmy.index)
poa_front = infinite_sheds.get_irradiance_poa(
                surface_tilt=orientation["surface_tilt"],
                surface_azimuth=orientation["surface_azimuth"],
                solar_zenith=solpos["apparent_zenith"],
                solar_azimuth=solpos["azimuth"],
                gcr=0.5,
                height=4,
                pitch=4,
                dni=tmy["dni"],
                ghi=tmy["ghi"],
                dhi=tmy["dhi"],
                albedo=0.3,
                model='haydavies',
                dni_extra=dni_extra,
                iam=1.0,
                npoints=100,
                vectorize=True)
poa_front.sum()
```

v0.9.5
```python
poa_global            1.606731e+06
poa_direct            1.222068e+06
poa_diffuse           3.846631e+05
poa_ground_diffuse    1.000573e+04
poa_sky_diffuse       3.746573e+05
```

v0.10.0
```python
poa_global            1.587275e+06
poa_direct            1.222068e+06
poa_diffuse           3.652078e+05
poa_ground_diffuse    7.194419e+03
poa_sky_diffuse       3.580134e+05
```

Here we get 39.1% difference in poa_ground_diffuse, 4.6% difference in poa_sky_diffuse which result in 1.2% difference on poa_global.

**Comment by mikofski** [comment]
This is expected, and pvlib-0.10 is more accurate. Maybe the release notes should say by a percentage point to better set expectations. Please review the pull request and issues closed to better understand what was changed and why. Also with GCR=0.5 (a bit high) you may see more differences than a lower GCR. I think we tested impact with lower GCR (0.3), lower height (1-m) and smaller albedo(0.2) as well. The effect of corrections on pvlib-0.10 may vary as the input parameters change?

**Reply by pasquierjb** [reply]
Agree that pvlib-0.10 is more accurate and thanks for the corrections. I tried with lower GCR, height and albedo on a fixed tilt system and got a 1.4% change on POA global. I just want to point out that for users that use pvlib in the industry, especially for non-conventional PV plants that use an important share of diffuse radiation (vertical bifacial system for instance), this might lead to differences up to 10% in their power estimation and the release note could be more specific.


```python
from pvlib.solarposition import get_solarposition
from pvlib.iotools.pvgis import get_pvgis_tmy
from pvlib.irradiance import get_extra_radiation
from pvlib import tracking
from pvlib.bifacial import infinite_sheds


lat, lon =  45.52, -122.67 # Portland
tmy = get_pvgis_tmy(lat, lon, map_variables=True)[0]
tmy.index = tmy.index.map(lambda dt: dt.replace(year=1990))
solpos = get_solarposition(tmy.index, lat, lon)


dni_extra = get_extra_radiation(tmy.index)
poa_front = infinite_sheds.get_irradiance_poa(
                surface_tilt=45,
                surface_azimuth=180,
                solar_zenith=solpos["apparent_zenith"],
                solar_azimuth=solpos["azimuth"],
                gcr=0.3,
                height=1,
                pitch=4,
                dni=tmy["dni"],
                ghi=tmy["ghi"],
                dhi=tmy["dhi"],
                albedo=0.2,
                model='haydavies',
                dni_extra=dni_extra,
                iam=1.0,
                npoints=100,
                vectorize=True)
poa_front.sum()
```

**Reply by kandersolar** [reply]
The difference certainly depends on the parameters.  In general I would expect the difference to increase with steeper tilt and higher GCR, since that's where the view factor calculations (which are what were updated between these two versions) are most relevant. 

Thanks @pasquierjb for highlighting that we may have understated the difference in the "what's new" page for some cases.  Too late to change the docs at https://pvlib-python.readthedocs.io/en/v0.10.1/, but we could consider revising that note for `latest` and the future `stable`.

**Comment by jules-ch** [comment]
To better understand the changes, view factor and diffuse factors are calculated with analytical solutions now instead of numerical discrete implementation that was wrong in the first place ?

**Reply by kandersolar** [reply]
Yes.  To elaborate: the infinite sheds model requires calculation of several view factors, with the ones relevant here being module -> sky and module -> ground.  These VFs vary with position on the module, so the infinite sheds model calculates the averages across the module surface using integrals.  In v0.9.5, those integrals are performed numerically by evaluating the VF at many discrete points.  However, @mikofski found that the VFs were calculated incorrectly, meaning the wrong quantity was being integrated.  Fixing those two VFs is the first change in v0.10.0.  The second change is that @cwhanse found that the corrected VFs could be integrated analytically, allowing us to replace the slower and marginally less accurate numerical integrations with closed-form calculations.
